src/compiler/parse.rs

```diff
@@ -520,7 +524,7 @@ impl Parser {
         left: Spanned<AST>
     ) -> Result<Spanned<AST>, Syntax> {
         self.consume(op)?;
-        let right = self.expression(prec.associate_left(), false)?;
```

Comment:
a bit more verbose, but better than the alternative, I think this might need to be refactored out further.